dependency_cache.go

```diff
@@ -187,7 +188,7 @@ func (d *DependencyCache) Artifact(dependency BuildpackDependency, mods ...Reque
 	}
 
 	if !isBinding {
-		d.overrideDependencySource(urlP)
+		d.setDependencyMirror(urlP)
```

Comment:
I think we should log a warning if there are both dependency mapping bindings and the dependency mirror setting being invoked at the same time.

Something like: `Both BP_DEPENDENCY_MIRROR and dependency bindings are present. BP_DEPENDENCY_MIRROR is being ignored. Please remove dependency map bindings if you wish to use BP_DEPENDENCY_MIRROR.`.